options_test.go

```diff
@@ -15,54 +15,66 @@ import (
 func TestEnvOptionParsing(t *testing.T) {
 	t.Run("string", func(t *testing.T) {
 		const val = "setup.sh"
-		t.Setenv("SETUP_SCRIPT", val)
+		t.Setenv(envbuilder.WithEnvPrefix("SETUP_SCRIPT"), val)
 		o := runCLI()
 		require.Equal(t, o.SetupScript, val)
 	})
 
 	t.Run("int", func(t *testing.T) {
-		t.Setenv("CACHE_TTL_DAYS", "7")
+		t.Setenv(envbuilder.WithEnvPrefix("CACHE_TTL_DAYS"), "7")
 		o := runCLI()
 		require.Equal(t, o.CacheTTLDays, int64(7))
 	})
 
 	t.Run("string array", func(t *testing.T) {
-		t.Setenv("IGNORE_PATHS", "/var,/temp")
+		t.Setenv(envbuilder.WithEnvPrefix("IGNORE_PATHS"), "/var,/temp")
 		o := runCLI()
 		require.Equal(t, o.IgnorePaths, []string{"/var", "/temp"})
 	})
 
 	t.Run("bool", func(t *testing.T) {
 		t.Run("lowercase", func(t *testing.T) {
-			t.Setenv("SKIP_REBUILD", "true")
-			t.Setenv("GIT_CLONE_SINGLE_BRANCH", "false")
+			t.Setenv(envbuilder.WithEnvPrefix("SKIP_REBUILD"), "true")
+			t.Setenv(envbuilder.WithEnvPrefix("GIT_CLONE_SINGLE_BRANCH"), "false")
 			o := runCLI()
 			require.True(t, o.SkipRebuild)
 			require.False(t, o.GitCloneSingleBranch)
 		})
 
 		t.Run("uppercase", func(t *testing.T) {
-			t.Setenv("SKIP_REBUILD", "TRUE")
-			t.Setenv("GIT_CLONE_SINGLE_BRANCH", "FALSE")
+			t.Setenv(envbuilder.WithEnvPrefix("SKIP_REBUILD"), "TRUE")
+			t.Setenv(envbuilder.WithEnvPrefix("GIT_CLONE_SINGLE_BRANCH"), "FALSE")
 			o := runCLI()
 			require.True(t, o.SkipRebuild)
 			require.False(t, o.GitCloneSingleBranch)
 		})
 
 		t.Run("numeric", func(t *testing.T) {
-			t.Setenv("SKIP_REBUILD", "1")
-			t.Setenv("GIT_CLONE_SINGLE_BRANCH", "0")
+			t.Setenv(envbuilder.WithEnvPrefix("SKIP_REBUILD"), "1")
+			t.Setenv(envbuilder.WithEnvPrefix("GIT_CLONE_SINGLE_BRANCH"), "0")
 			o := runCLI()
 			require.True(t, o.SkipRebuild)
 			require.False(t, o.GitCloneSingleBranch)
 		})
 
 		t.Run("empty", func(t *testing.T) {
-			t.Setenv("GIT_CLONE_SINGLE_BRANCH", "")
+			t.Setenv(envbuilder.WithEnvPrefix("GIT_CLONE_SINGLE_BRANCH"), "")
 			o := runCLI()
 			require.False(t, o.GitCloneSingleBranch)
 		})
 	})
+
+	t.Run("legacy", func(t *testing.T) {
+		legacyEnvValue := "./setup-legacy-script.sh"
+		t.Setenv("SETUP_SCRIPT", legacyEnvValue)
+		o := runCLI()
+		require.Equal(t, o.SetupScript, legacyEnvValue)
+
+		envValue := "./setup-script.sh"
+		t.Setenv(envbuilder.WithEnvPrefix("SETUP_SCRIPT"), envValue)
+		o = runCLI()
+		require.Equal(t, o.SetupScript, envValue)
+	})
```

Comment:
@mtojek Idk what is the best way to test the legacy support for the options but I just thought about getting one of the options and testing the values with the prefix and without it. Wdyt?

Reply:
@johnstcn answered my question here https://github.com/coder/envbuilder/pull/180#discussion_r1596967496.